optimize.py

```diff
@@ -195,13 +198,74 @@ def _check_missing_players(all_players, min_cost, e_raise):
                                                 str(total_report))
         raise Exception('Total missing players at price point: ' + str(miss))
 
+def create_upload_file_and_map_pids(pid_file, test_mode=False):
+    # create upload file
+    csv_name = 'test' if test_mode else 'current'
+    with open(fnu.format(csv_name), "w") as f:
+        writer = csv.writer(f)
+        writer.writerow(['QB','RB','RB', 'WR', 'WR', 'WR', 'TE', 'FLEX', 'DST'])
+
+    # create a player map from name/position to id
+    player_map = {}
+    with open(pid_file, "r") as f:
+        n = 0
+        for line in f.readlines():
+            n += 1
+            if 'TeamAbbrev' in line: # line with field names was found
+                fields= line.split(',')
+                break
+        f.close()
+        f = islice(open(pid_file, "r"), n, None)
+        reader = csv.DictReader(f, fieldnames=fields)
+        for line in reader:
+            player_map[line[' Name'] + " " + line['Position']] = line[' ID']
+    return player_map
+
+def update_upload_csv(player_map, players, test_mode=False):
+    csv_name = 'test' if test_mode else 'current'
+    with open(fnu.format(csv_name), "a") as f:
+        writer = csv.writer(f)
+        player_ids = []
+        flexid=-1
+        rbs=0
+        wrs=0
+
+        for p in players:
+            # For the players in the roster, get their id.
+            # they are sorted when they are printed out, so the order is good, excepet
+            # for the flex player
+            id = player_map[p.name + " " + p.pos]
```

Comment:
Minor, but I always try to avoid overwriting `id`since it's a built in function in Python.